src/api/api.ts

```diff
@@ -133,150 +148,132 @@ export function getEmptyCheckTypes(): Record<string, SpecRaw> {
+        new Date(check.metadata.creationTimestamp) > new Date(checkByType[type].metadata.creationTimestamp ?? 0)
+      ) {
+        checkByType[type] = check;
+      }
     }
-  }
-
-  return checksBySeverity;
-}
-
-export async function getLastChecks(): Promise<CheckRaw[]> {
-  const checkByType: Record<string, CheckRaw> = {};
-  const checks = await getChecks();
 
-  for (const check of checks) {
-    const type = check.metadata.labels?.[CHECK_TYPE_LABEL];
+    return Object.values(checkByType);
+  }, [data]);
 
-    if (!type) {
-      // No type found for check under "check.metadata.labels[advisor.grafana.app/type]", skipping.
-      continue;
-    }
-
-    if (!check.metadata.creationTimestamp) {
-      // Empty creationTimestamp for check at "check.metadata.creationTimestamp", skipping.
-      continue;
-    }
+  return { checks, ...listChecksState };
+}
 
-    if (check.metadata.annotations?.[STATUS_ANNOTATION] !== 'processed') {
-      // Check is not processed yet, skipping.
-      continue;
-    }
+export function useCreateChecks() {
+  const { checkTypes } = useCheckTypes();
+  const [createCheck, createCheckState] = useCreateCheckMutation();
 
-    // If the check is the first one of its type, store it
-    if (!checkByType[type]) {
-      checkByType[type] = check;
-      continue;
+  const createChecks = React.useCallback(() => {
+    if (!checkTypes) {
+      return;
     }
-
-    const prevTimestamp = new Date(checkByType[type].metadata.creationTimestamp ?? 0);
-    const currentTimestamp = new Date(check.metadata.creationTimestamp);
-
-    if (currentTimestamp > prevTimestamp) {
-      checkByType[type] = check;
+    for (const type of checkTypes) {
+      createCheck({
+        check: {
+          kind: 'Check',
+          apiVersion: 'advisor.grafana.app/v0alpha1',
+          spec: { data: {} },
+          metadata: {
+            generateName: 'check-',
+            labels: { 'advisor.grafana.app/type': type.metadata.name ?? '' },
+            namespace: config.namespace,
+          },
+          status: { report: { count: 0, failures: [] } },
+        },
+      });
     }
-  }
+  }, [createCheck, checkTypes]);
 
-  return Object.values(checkByType);
+  return { createChecks, createCheckState };
 }
 
-export async function getChecks(): Promise<CheckRaw[]> {
-  const response = await checkClient.list();
+export function useDeleteChecks() {
+  const [deleteCheckMutation, deleteChecksState] = useDeleteCheckMutation();
+  const deleteChecks = () => deleteCheckMutation({ name: '' });
 
-  return response.data.items;
+  return { deleteChecks, deleteChecksState };
 }
 
-export async function getCheck(name: string): Promise<CheckRaw> {
-  const response = await checkClient.get(name);
+function useIncompleteChecks(names?: string[]) {
+  const listChecksState = useListCheckQuery(
+    {},
+    {
+      refetchOnMountOrArgChange: true,
+      pollingInterval: 0, // We'll handle polling in useCompletedChecks
```

Comment:
I'm sure there's a good reason for this but why not use the internal rtk query polling?

Reply:
Mostly because it was how it worked before but this may be niced.  Let me try.